pkg/cache/multi_namespace_cache.go

```diff
@@ -131,6 +189,10 @@ func (c *multiNamespaceCache) IndexField(ctx context.Context, obj client.Object,
 			return err
 		}
 	}
+
+	if err := c.clusterCache.IndexField(ctx, obj, field, extractValue); err != nil {
```

Comment:
I think this also needs to be called only if obj is namespaced, otherwise iterate through `c.namespaceToCache`.